diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42dc198e..d9323f47 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@ else()
   # the code below can be removed; e.g. pybind needs Interpreter and Development components
   # see https://pybind11.readthedocs.io/en/stable/cmake/index.html#new-findpython-mode
   if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
-    set(IGN_PYTHON_VERSION "3")
+    set(IGN_PYTHON_VERSION "3.12")
     find_package(PythonInterp ${IGN_PYTHON_VERSION} QUIET)
     if(PYTHONINTERP_FOUND)
       set(Python3_FOUND ${PYTHONINTERP_FOUND})
@@ -111,7 +111,7 @@ else()
       set(Python3_VERSION ${PYTHONLIBS_VERSION_STRING})
     endif()
   else()
-    find_package(Python3 QUIET COMPONENTS Interpreter Development)
+    find_package(Python3 3.12 QUIET COMPONENTS Interpreter Development)
   endif()
 
   if (NOT Python3_FOUND)
